### PR TITLE
 [RELEASE-1.30.2] [SRVCOM-2744] Disable http2 in webhooks

### PR DESCRIPTION
**Pull request by skonto**
Fixes JIRA #SRVCOM-2744

## Proposed Changes

- Similar to #2345 
- Unlike the other PRs we can't use tls options for the ctrl manager webhook as it is not available.

### CODE REVIEW

File: vendor/sigs.k8s.io/controller-runtime/pkg/webhook/server.go [hunk 1]
```diff
@@ -232,7 +232,7 @@ func (s *Server) Start(ctx context.Context) error {
 	}
 
 	cfg := &tls.Config{ //nolint:gosec
-		NextProtos:     []string{"h2"},
+		NextProtos:     []string{"http/1.1"},
```
Comment (skonto) [inline vendor/sigs.k8s.io/controller-runtime/pkg/webhook/server.go:235]: fix is in

**Comment by skonto** [comment]
lint:
```
  level=error msg="Running error: 1 error occurred:\n\t* can't run linter goanalysis_metalinter: buildir: failed to load package featuressteps: could not load export data: no export data for \"knative.dev/eventing-kafka-broker/test/rekt/features/featuressteps\"\n\n"
```

File: go.mod [hunk 2]
```diff
@@ -139,13 +139,13 @@ require (
 	go.opencensus.io v0.24.0 // indirect
 	go.uber.org/automaxprocs v1.4.0 // indirect
 	go.uber.org/multierr v1.8.0 // indirect
-	golang.org/x/crypto v0.11.0 // indirect
+	golang.org/x/crypto v0.14.0 // indirect
 	golang.org/x/mod v0.8.0 // indirect
 	golang.org/x/net v0.12.0 // indirect
```
[the remark this answers is not in the document]
Comment (ReToCode) [reply go.mod:144]: Or if not, can we skip bumping the dependencies at all here (as it introduces a bit more risk, as it is a patch release)?

[the remark this answers is not in the document]
Comment (skonto) [reply go.mod:144]: Yeah probably it is not updated?

**Comment by ReToCode** [comment]
@pierDipi @matzew can you check if eventing + reconciler-test bumps are ok?

File: go.mod (hunk 2 above)
[the remark this answers is not in the document]
Comment (ReToCode) [reply go.mod:144]: Yes it is not upstream, but do we need if if we disable http2 anyway?

[the remark this answers is not in the document]
Comment (skonto) [reply go.mod:144]: I dont think we need it but we could bump it manually.
Btw as you mentioned midstream for 1.10 it is there directly: https://github.com/openshift-knative/serving/blob/release-v1.10/go.mod#L27.

[the remark this answers is not in the document]
Comment (skonto) [reply go.mod:144]: I updated that dep anyway.

[the remark this answers is not in the document]
Comment (ReToCode) [reply go.mod:144]: We probably should change as little as possible. So maybe reverting the deps bump and only do `go mod replace` for the `x/net` thing?

[the remark this answers is not in the document]
Comment (skonto) [reply go.mod:144]: I don't mind if we dont feel confident I can revert.

[the remark this answers is not in the document]
Comment (ReToCode) [reply go.mod:144]: @pierDipi , @matzew , @mgencur opinions? ^^

**Comment by pierDipi** [comment]
@ReToCode it seems there is no patch anymore for them?

**Comment by ReToCode** [comment]
> @ReToCode it seems there is no patch anymore for them?

Yes sorry about that, we removed it for 1.30.2 as the risk is a bit high for a patch-release.

**Review by ReToCode — APPROVED** [review]
/lgtm
/approve

**Pull request merged**